pyproject.toml

```diff
@@ -35,16 +35,18 @@ huggingface-hub = ">0.23.2"
 # we need to support both versions
 numpy = ">=1.26.0"
 
+[tool.poetry.group.server.dependencies]
+fastapi = "^0.115.8"
+pydantic = "^2.10.6"
+pyyaml = "^6.0.2"
```

Comment:
Sigh. I really hate this top-version by poetry. Does it means that pyyaml 6.0.1 will not work? 5.4 will not work?

Reply:
Caret constraints are somewhat more complicated than explicit `><=` but essentially it's just a short hand for the same. In case of pyyaml, `>=6.0.2,<7.0.0`. 

https://python-poetry.org/docs/dependency-specification/#caret-requirements

Reply:
I know the caret operator. I was arguing if there can be:

* ^6.0.1
* or >5.4
* or >1.0

I think that we (as developers) should spend time investigating what versions we really need. And not use the latest version that is available.